### PR TITLE
DAOS-3203 rebuild: add time costs to rebuild

### PR DESCRIPTION
**Pull request by wangdi1**
Add time costs to rebuild query information.

Signed-off-by: Wang Di <di.wang@intel.com>

### CODE REVIEW

File: src/rebuild/README.md [hunk 1]
```diff
@@ -192,8 +192,8 @@ which will return its rebuild status to client as well.
 struct daos_rebuild_status {
         /** pool map version in rebuilding or last completed rebuild */
         uint32_t                rs_version;
-        /** padding bytes */
-        uint32_t                rs_pad_32;
+	/** Time (Seconds) for the rebuild */
+        uint32_t                rs_seconds;
```
Comment (liuxuezhao) [inline src/rebuild/README.md]: other changed/added fields as well?

File: src/rebuild/srv.c [hunk 2]
```diff
@@ -1693,13 +1702,16 @@ rebuild_tgt_status_check(void *arg)
...
+			iv.riv_size = status.size;
 		} else {
 			iv.riv_toberb_obj_count = rpt->rt_toberb_objs -
 						  rpt->rt_reported_toberb_objs;
 			iv.riv_obj_count = status.obj_count -
 					   rpt->rt_reported_obj_cnt;
 			iv.riv_rec_count = status.rec_count -
 					   rpt->rt_reported_rec_cnt;
+			iv.riv_size = status.size -
+					   rpt->rt_reported_size;
```
Comment (liuxuezhao) [inline src/rebuild/srv.c]: seems need to rebase to latest master as there are a few refining at above L1698 ~ L1700, may need to add below to there:
if (status.size < rpt->rt_reported_size)
status.size = rpt->rt_reported_size;

Comment (wangdi1) [reply src/rebuild/srv.c]: ah, ok

File: src/rebuild/README.md (hunk 1 above)
Comment (wangdi1) [reply src/rebuild/README.md]: ok

File: src/tests/ftest/util/test_utils.py [hunk 3]
```diff
@@ -345,24 +345,26 @@ def check_pool_daos_space(self, s_total=None, s_free=None):
...
             rs_errno (int, optional): rebuild error number. Defaults to None.
             rs_done (int, optional): rebuild done flag. Defaults to None.
+            rs_fail_rank (int, optional): rebuild fail target. Defaults to None.
             rs_toberb_obj_nr (int, optional): number of objects to be rebuilt.
                 Defaults to None.
             rs_obj_nr (int, optional): number of rebuilt objects.
                 Defaults to None.
             rs_rec_nr (int, optional): number of rebuilt records.
                 Defaults to None.
+	    rs_size (int, optional): size of all rebuilt records.
```
Comment (phender) [inline src/tests/ftest/util/test_utils.py]: The spacing appears off on this line

File: src/tests/ftest/util/test_utils.py [hunk 4]
```diff
@@ -345,24 +345,26 @@ def check_pool_daos_space(self, s_total=None, s_free=None):
             for index, item in enumerate(val)]
         return self._check_info(checks)
 
-    def check_rebuild_status(self, rs_version=None, rs_pad_32=None,
+    def check_rebuild_status(self, rs_version=None, rs_seconds=None,
```
Comment (phender) [inline src/tests/ftest/util/test_utils.py]: Is there any reason to not include the `rs_padding32` item?

File: src/tests/ftest/util/test_utils.py [hunk 5]
```diff
@@ -336,24 +336,26 @@ def check_pool_daos_space(self, s_total=None, s_free=None):
             for index, item in enumerate(val)]
         return self._check_info(checks)
 
-    def check_rebuild_status(self, rs_version=None, rs_pad_32=None,
-                             rs_errno=None, rs_done=None,
-                             rs_toberb_obj_nr=None, rs_obj_nr=None,
-                             rs_rec_nr=None):
+    def check_rebuild_status(self, rs_version=None, rs_seconds=None,
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/util/test_utils.py]: This error can be ignored by adding the following line between lines 352 and 353:
```
        # pylint: disable=too-many-arguments
```

File: src/tests/ftest/util/test_utils.py (hunk 3 above)
Comment (wangdi1) [reply src/tests/ftest/util/test_utils.py]: sorry, my bad.

File: src/tests/ftest/util/test_utils.py (hunk 4 above)
Comment (wangdi1) [reply src/tests/ftest/util/test_utils.py]: just missing that.

**Comment by johannlombardi** [comment]
please address the conflict so that i can land.
src/utils/py/daos_cref.py has been moved to src/client/pydaos/raw.

Thanks

**Pull request merged**